.githooks/pre-commit

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env sh
+":" //; if command -v pwsh >/dev/null 2>&1; then pwsh -ExecutionPolicy Bypass -File .githooks/pre-commit.ps1; else sh .githooks/pre-commit.sh; fi; exit $?     # Try PowerShell Core first, then sh on Unix
```

Comment:
The script assumes that the commit operation is executed from the repository's root directory. It uses relative paths like `.githooks/xxx` and `.githooks\xxx` to locate the hook scripts. If the commit command is run from a subdirectory, these paths will not resolve correctly, leading to errors because the scripts cannot be found.

Reply:
Actually it can work in the subdirectory 
![image](https://github.com/user-attachments/assets/76829d08-a34c-4fba-8f73-ba52fcb70853)